config.go

```diff
@@ -0,0 +1,324 @@
+	MinDiskSpaceMB uint64 `mapstructure:"min-disk-space-mb"`
+	MinSecs        int    `mapstructure:"min-secs"`
+	MaxSecs        int    `mapstructure:"max-secs"`
+	PreviewSecs    int    `mapstructure:"preview-secs"`
+}
+
+var DefaultThermalRecorder = ThermalRecorder{
+	MaxSecs:        600,
+	MinSecs:        10,
+	PreviewSecs:    3,
+	MinDiskSpaceMB: 200,
+	OutputDir:      "/var/spool/cptv",
+}
+
+type ThermalMotion struct {
+	DynamicThreshold bool   `mapstructure:"min-secs"`
+	TempThresh       uint16 `mapstructure:"temp-thresh"`
+	DeltaThresh      uint16 `mapstructure:"delta-thresh"`
+	CountThresh      int    `mapstructure:"count-thresh"`
+	FrameCompareGap  int    `mapstructure:"frame-compare-gap"`
+	UseOneDiffOnly   bool   `mapstructure:"use-one-diff-only"`
+	TriggerFrames    int    `mapstructure:"trigger-frames"`
+	WarmerOnly       bool   `mapstructure:"warmer-only"`
+	EdgePixels       int    `mapstructure:"edge-pixels"`
+	Verbose          bool   `mapstructure:"verbose"`
+}
+
+var DefaultThermalMotion = ThermalMotion{
+	DynamicThreshold: true,
+	TempThresh:       2900,
+	DeltaThresh:      50,
+	CountThresh:      3,
+	FrameCompareGap:  45,
+	Verbose:          false,
+	TriggerFrames:    2,
+	UseOneDiffOnly:   true,
+	WarmerOnly:       true,
+	EdgePixels:       1,
+}
+
+type ThermalThrottler struct {
+	Activate   bool
+	BucketSize time.Duration `mapstructure:"bucket-size"`
+	MinRefill  time.Duration `mapstructure:"min-refill"`
+}
+
+var DefaultThermalThrottler = ThermalThrottler{
+	Activate:   true,
+	BucketSize: 10 * time.Minute,
+	MinRefill:  10 * time.Minute,
+}
+
+type Ports struct {
+	Managementd int
+}
+
+var DefaultPorts = Ports{
+	Managementd: 80,
+}
+
+type Secrets struct {
+	DevicePassword string `mapstructure:"device-password"`
+}
+
+type GPIO struct {
+	ThermalCameraPower string `mapstructure:"thermal-camera-power"`
+	ModemPower         string `mapstructure:"modem-power"`
+}
+
+var DefaultGPIO = GPIO{
+	ThermalCameraPower: "GPIO23",
+	ModemPower:         "GPIO22",
+}
+
+const (
+	DeviceKey           = "device"
+	WindowsKey          = "windows"
+	LocationKey         = "location"
+	TestHostsKey        = "test-hosts"
+	BatteryKey          = "battery"
+	ModemdKey           = "modemd"
+	LeptonKey           = "lepton"
+	ThermalRecorderKey  = "thermal-recorder"
+	ThermalMotionKey    = "thermal-motion"
+	ThermalThrottlerKey = "thermal-throttler"
+	PortsKey            = "ports"
+	SecretsKey          = "secrets"
+	GPIOKey             = "gpio"
+
+	DefaultConfigDir = "/etc/cacophony"
+	configFileName   = "config.toml"
+	lockRetryDelay   = 678 * time.Millisecond
+)
+
+// Helpers for testign purposes
+var fs = afero.NewOsFs()
+var now = time.Now
+var lockFilePath = func(configFile string) string {
+	return configFile + ".lock"
+}
+var lockTimeout = 10 * time.Second
+
+// New created a new config and loads files from the given directory
+func New(dir string) (*Config, error) {
+	// TODO Take service name and restart service if config changes
```

Comment:
Why would you take the service name? shouldn't there be a service associated with each area of the config?

Reply:
I was thinking when creating a New `Config` you can pass the name of the service that is calling it and then each time `Unmarshal` (and maybe `Get`) was called you add the section key to the `accessedSections` map and then pass that updated `accessedSections` map (set) to a service and that service is watching for the changes in the config file, restarting services when needed.
This means you don't need to update/write all the services and what sections they use. Simply by using them they get added.